modules/compute/vm-instance/variables.tf

```diff
@@ -73,6 +73,15 @@ variable "name_prefix" {
   default     = null
 }
 
+variable "always_include_deployment_name" {
```

Comment:
nit (please feel free to ignore):
`always_include_deployment_name` could be more intuitive
I explain:  the default behavior is to have prefix OR deployment name. 
By setting `always_include_deployment_name`, we are now adding deployment name and then the prefix. Would it be simpler to call it something like `add_deployment_name_before_prefix`?

Reply:
Discussed with @tpdownes who found either name acceptable. Updated to this suggested name.